sklearn/cluster/k_means_.py

```diff
@@ -441,10 +442,19 @@ def _kmeans_single(X, n_clusters, x_squared_norms, max_iter=300,
             best_centers = centers.copy()
             best_inertia = inertia
 
-        if squared_norm(centers_old - centers) <= tol:
+        shift = squared_norm(centers_old - centers)
+        if shift <= tol:
             if verbose:
                 print("Converged at iteration %d" % i)
+
             break
+
+        if shift > 0:
```

Comment:
It could be zero if the centers didn't shift.
Not sure if we should have an eps here. @vighneshbirodkar can you check that with normal convergence this if is not true?